### PR TITLE
test: provide coverage for a couple of recent-ish issues

### PR DESCRIPTION
**Pull request by mrc0mmand**

### CODE REVIEW

**Review by yuwata — APPROVED** [review]

File: test/units/testsuite-73.sh [hunk 1]
```diff
@@ -648,6 +648,40 @@ EOF
     assert_in "XKBLAYOUT=us" "$output"
 }
 
+locale_gen_cleanup() {
+    if mountpoint -q /usr/lib/locale; then
+        rm -fr /usr/lib/locale/*
```
Comment (yuwata) [inline test/units/testsuite-73.sh]: Hm? Why necessary?

Comment (mrc0mmand) [reply test/units/testsuite-73.sh]: Various running apps keep the mountpoint busy:

```
++ locale_gen_cleanup
++ mountpoint -q /usr/lib/locale
++ umount /usr/lib/locale
umount: /usr/lib/locale: target is busy.
```

```
# lsof | grep /usr/lib/locale
agetty    288                          root  mem       REG              254,2              574367 /usr/lib/locale/locale-archive (path dev=0,39, inode=3)
agetty    288                          root  mem       REG              254,2              527694 /usr/lib/locale/C.UTF-8/LC_CTYPE (stat: No such file or directory)
bash      544                       vagrant  mem       REG              254,2              574367 /usr/lib/locale/locale-archive (path dev=0,39, inode=3)
sudo      549                          root  mem       REG              254,2              574367 /usr/lib/locale/locale-archive (path dev=0,39, inode=3)
su        550                          root  mem       REG              254,2              574367 /usr/lib/locale/locale-archive (path dev=0,39, inode=3)
bash      551                          root  mem       REG              254,2              574367 /usr/lib/locale/locale-archive (path dev=0,39, inode=3)
lsof      748                          root  mem       REG               0,39  3415040          3 /usr/lib/locale/locale-archive
grep      749                          root  mem       REG               0,39  3415040          3 /usr/lib/locale/locale-archive
lsof      750                          root  mem       REG               0,39  3415040          3 /usr/lib/locale/locale-archive
```

Removing the stuff before the umount call seems to help (albeit it's not very pretty).

Comment (mrc0mmand) [reply test/units/testsuite-73.sh]: Also, lazy unmount seems to work as well, which might be a bit prettier.

File: test/units/testsuite-07.mount-invalid-chars.sh [hunk 2]
```diff
@@ -0,0 +1,70 @@
+#!/usr/bin/env bash
+# SPDX-License-Identifier: LGPL-2.1-or-later
+set -eux
+set -o pipefail
+
+# Don't send invalid characters over dbus if a mount contains them
+
+at_exit() {
+    mountpoint -q /proc/1/mountinfo && umount /proc/1/mountinfo
+    [[ -e /tmp/fstab.bak ]] && mv -f /tmp/fstab /etc/fstab
```
Comment (dtardon) [inline test/units/testsuite-07.mount-invalid-chars.sh:10]: The source file for `mv` is missing an extension.

File: test/units/testsuite-73.sh [hunk 3]
```diff
@@ -656,11 +656,18 @@ EOF
 export SYSTEMD_KBD_MODEL_MAP=/usr/lib/systemd/tests/testdata/test-keymap-util/kbd-model-map
 
 enable_debug
-test_locale
-test_vc_keymap
-test_x11_keymap
-test_convert
-test_validate
+
+# Create a list of all functions prefixed with testcase_
```
Comment (dtardon) [inline test/units/testsuite-73.sh:692]: This looks like something that should be wrapped into a function (`run_testcases`?), moved into `util.sh` and used in other tests too (`TEST-35-LOGIN` comes to mind).

Btw, is there any particular reason for the chosen prefix? It seems to me that `test_` would work just as well...

File: test/units/testsuite-07.mount-invalid-chars.sh (hunk 2 above)
Comment (mrc0mmand) [reply test/units/testsuite-07.mount-invalid-chars.sh:10]: Oops, addressed in https://github.com/systemd/systemd/pull/27733.

File: test/units/testsuite-73.sh (hunk 3 above)
Comment (mrc0mmand) [reply test/units/testsuite-73.sh:692]: `test_` is way too generic and might match possible test-related helpers (if introduced in the future).

**Pull request merged**